zeno/agents/kba/tool_kba_timeseries.py

```diff
@@ -0,0 +1,192 @@
+    column: constr(min_length=1) = Field(
+        ..., description="Name of the column/metric being analyzed"
+    )
+    title: constr(min_length=5) = Field(
+        ..., description="Title describing the time series insight"
+    )
+    description: constr(min_length=20) = Field(
+        ...,
+        description="Detailed explanation of the time series pattern or finding",
+    )
+    data: List[TimeSeriesPoint] = Field(
+        ...,
+        min_items=2,  # Require at least 2 points for a time series
+        description="Time series data points",
+    )
+    trend_direction: Optional[str] = Field(
+        None,
+        description="Overall trend direction (increasing, decreasing, stable)",
+    )
+    max_change: Optional[dict] = Field(
+        None,
+        description="Details about the maximum change between consecutive periods",
+    )
+
+    @validator("data")
+    def validate_data_ordering(cls, v):
+        # Ensure years are in chronological order
+        years = [point.year for point in v]
+        if years != sorted(years):
+            raise ValueError(
+                "Time series points must be in chronological order"
+            )
+
+        # Check for duplicate years
+        if len(years) != len(set(years)):
+            raise ValueError("Duplicate years found in time series")
+
+        return v
+
+    @validator("data")
+    def calculate_trend(cls, v, values):
+        # Calculate trend if not provided
+        if len(v) >= 2:
+            start_value = v[0].value
+            end_value = v[-1].value
+            if (
+                "trend_direction" not in values
+                or values["trend_direction"] is None
+            ):
+                if end_value > start_value * 1.05:  # 5% threshold
+                    values["trend_direction"] = "increasing"
+                elif end_value < start_value * 0.95:  # 5% threshold
+                    values["trend_direction"] = "decreasing"
+                else:
+                    values["trend_direction"] = "stable"
+
+            # Calculate max change if not provided
+            if "max_change" not in values or values["max_change"] is None:
+                max_change = 0
+                max_change_years = (0, 0)
+                for i in range(1, len(v)):
+                    change = abs(v[i].value - v[i - 1].value)
+                    if change > max_change:
+                        max_change = change
+                        max_change_years = (v[i - 1].year, v[i].year)
+
+                values["max_change"] = {
+                    "value": max_change,
+                    "period": f"{max_change_years[0]}-{max_change_years[1]}",
+                }
+
+        return v
+
+
+class TimeSeriesResponse(BaseModel):
+    insights: List[TimeSeriesInsight] = Field(
+        ...,
+        min_items=1,
+        description="List of time series insights from the analysis",
+    )
+
+
+class ColumnSelectionOutput(BaseModel):
```

Comment:
Consider importing the column selection logic from other agent instead of duplicating?